i3config syntax with better keyword/type distinction

litoj opened the pull request:
This syntax uses a couple more colour highlights to match the actual function of used keywords in the config file. This way it is easier to figure out what does what.
- distinctly highlights key combos to clearly differentiate it from the rest of the command
- all keywords that can be used as command line commands are highlighted as Functions
- conditions are now properly highlighted with all operators clearly visible
+ some bug fixes

file updated at: https://github.com/JosefLitos/vim-i3config

hiqua commented on runtime/syntax/i3config.vim:
```diff
@@ -1,9 +1,9 @@
-" Maintainer: Quentin Hibon (github user hiqua)
-" Version: 0.4
-" Last Change: 2022 Jun 05
+" Maintainer: Josef Litoš (github JosefLitos)
```
Sure about that? :smile:

hiqua commented on runtime/syntax/i3config.vim:
```diff
@@ -1,9 +1,9 @@
-" Version: 0.4
-" Last Change: 2022 Jun 05
+" Maintainer: Josef Litoš (github JosefLitos)
+" Version: 4.20
```
I guess we can remove the version line altogether, it makes little sense IMHO.

hiqua commented on runtime/syntax/i3config.vim:
```diff
@@ -17,39 +17,39 @@ endif
 scriptencoding utf-8
 
+" Error
+syn match i3ConfigError /.*/
```
I've removed that already in a separate commit, can you remove it as well? It poses problem whenever a new keyword is introduced e.g. in sway, and we're not fully up to date, which is bound to happen regularly.

hiqua reviewed (changes requested):
I know it's a long shot, but, before I deep-dive into the code to review it, any chance you could try to reduce the diff? That would make my life way easier.

Otherwise I'll make do with this, but it might take me more time.

litoj committed:
Update i3config.vim

litoj committed:
Update i3config.vim

litoj committed:
added 'show' for scratchpad

litoj committed:
Fixed multiline strings

litoj committed:
fixed general line continuation

chrisbra committed:
patch 9.0.0587: Unicode tables are outdated

Problem:    Unicode tables are outdated.
Solution:   Update to Unicode release 15. (Christian Brabandt, closes #11220)

sodero committed:
patch 9.0.0588: MorphOS build is broken

Problem:    MorphOS build is broken.
Solution:   Add "-lm" to LDFLAGS and "-noixemul" to CFLAGS. (Ola Söder,
            closes #11222)

sodero committed:
patch 9.0.0589: on AmigaOS4 the pid is available but the task address is used

Problem:    On AmigaOS4 the pid is available but the task address is used.
Solution:   Use getpid().  (Ola Söder, closes #11224)

brammool committed:
patch 9.0.0590: after exiting Insert mode spelling not checked in next line

Problem:    After exiting Insert mode spelling is not checked in the next
            line.
Solution:   When spelling is enabled redraw the next line after exiting Insert
            mode in case the spell highlight needs updating.

brammool committed:
patch 9.0.0591: message window popup shows on only one tab page

Problem:    Message window popup shows on only one tab page. (Naruhiko
            Nishino)
Solution:   Show the message window popup on all tab pages. (closes #11231)

brammool committed:
patch 9.0.0592: display not cleared when scrolling back in messages

Problem:    Display not cleared when scrolling back in messages, a background
            color is set and t_ut is empty.
Solution:   Clear to the end of the display if needed. (closes #8973)

sashashura committed:
patch 9.0.0593: CI actions have too many permissions

Problem:    CI actions have too many permissions.
Solution:   Restrict permissions to what is required. (closes #11223)

dundargoc committed:
patch 9.0.0594: Makefile error message causes a shell error

Problem:    Makefile error message causes a shell error.
Solution:   Put the message in single quotes. (closes #11232)

brammool committed:
patch 9.0.0595: extra newline in messages after a verbose shell message

Problem:    Extra newline in messages after a verbose shell message.
Solution:   Output the newline with msg_putchar_attr(). (closes #11233)
            Make it possible to filter a screendump before comparing it.

pheiduck committed:
patch 9.0.0596: CI on Mac M1 has the channel feature disabled

Problem:    CI on Mac M1 has the channel feature disabled.
Solution:   Include the channel feature.

regomne committed:
patch 9.0.0597: cannot close a tab page with the middle mouse button

Problem:    Cannot close a tab page with the middle mouse button.
Solution:   Support closing a tab page with the middle mouse button, like many
            other programs. (closes #10746)

brammool committed:
patch 9.0.0598: using negative array index with negative width window

Problem:    Using negative array index with negative width window.
Solution:   Make sure the window width does not become negative.

ObserverOfTime committed:
patch 9.0.0599: latexmkrc files are not recognized

Problem:    Latexmkrc files are not recognized.
Solution:   Use Perl filetype for latexmkrc files. (closes #11241)

ObserverOfTime committed:
patch 9.0.0600: GYP files are not recognized

Problem:    GYP files are not recognized.
Solution:   Recognize GYP files. (closes #11242)

yegappan committed:
patch 9.0.0601: too much indent

Problem:    Too much indent.
Solution:   Return out early from a funtion. (Yegappan Lakshmanan,
            close #11238)

rhysd committed:
patch 9.0.0602: new TypeScript extensions are not recognized

Problem:    New TypeScript extensions are not recognized.
Solution:   Recognize .mts and .cts files. (closes #11237)

luukvbaal committed:
patch 9.0.0603: with 'nosplitscroll' folds are not handled correctly

Problem:    With 'nosplitscroll' folds are not handled correctly.
Solution:   Take care of closed folds when moving the cursor. (Luuk van Baal,
            closes #11234)

ObserverOfTime committed:
patch 9.0.0604: luacheckrc file is not recognized

Problem:    Luacheckrc file is not recognized.
Solution:   Use lua filetype for luacheckrc. (closes #11236)

luukvbaal committed:
patch 9.0.0605: dump file missing

Problem:    Dump file missing.
Solution:   Add the missing dump file. (issue #11234)

brammool committed:
patch 9.0.0606: system() opens a terminal window when "!" is in 'guioptions'

Problem:    system() opens a terminal window when using the GUI and "!" is in
            'guioptions'.
Solution:   Do not use a terminal window when the SHELL_SILENT flag is used.
            (closes #11202)

brammool committed:
patch 9.0.0607: verbose echo message test fails on Mac OS

Problem:    Verbose echo message test fails on Mac OS.
Solution:   Skip the test on Mac OS.

brammool committed:
patch 9.0.0608: with spelling, deleting a full stop does not update next line

Problem:    With spell checking, deleting a full stop at the end of a line
            does not update SpellCap at the start of the next line.
Solution:   Update the next line when characters have been deleted.  Also when
            using undo.

brammool committed:
Update runtime files

brammool committed:
patch 9.0.0609: blockedit test fails because of wrong indent

Problem:    Blockedit test fails because of wrong indent.
Solution:   Adjust the expected text temporarily

brammool committed:
patch 9.0.0610: global interrupt test fails when run under valgrind

Problem:    Global interrupt test fails when run under valgrind.
Solution:   Use TermWait().

brammool committed:
patch 9.0.0611: tests delete files with a separate delete() call

Problem:    Tests delete files with a separate delete() call.
Solution:   Use deferred delete.

brammool committed:
patch 9.0.0612: blockedit test passes with wrong result

Problem:    Blockedit test passes with wrong result.
Solution:   Add a "vim9script" line to make indenting work.

dpelle committed:
patch 9.0.0613: running source tests leaves file behind

Problem:    Running source tests leaves file behind.
Solution:   Add the "D" flag to writefile(). (Dominique Pellé, closes #11243)
            Also do this for other writefile() calls and drop delete().

brammool committed:
patch 9.0.0614: SpellFileMissing autocmd may delete buffer

Problem:    SpellFileMissing autocmd may delete buffer.
Solution:   Disallow deleting the current buffer to avoid using freed memory.

brammool committed:
patch 9.0.0615: using reduce() on a list from range() is a bit slow

Problem:    Using reduce() on a list from range() is a bit slow.
Solution:   Avoid materializing the list.

brammool committed:
patch 9.0.0616: spell test fails because error message changed

Problem:    Spell test fails because error message changed.
Solution:   Adjust expected error message.

brammool committed:
patch 9.0.0617: calling function for reduce() has too much overhead

Problem:    Calling function for reduce() has too much overhead.
Solution:   Only call clear_tv() when needed.

brammool committed:
patch 9.0.0618: calling function for reduce() has too much overhead

Problem:    Calling function for reduce() has too much overhead.
Solution:   Do not create a funccall_T every time.

brammool committed:
patch 9.0.0619: too many delete() calls in tests

Problem:    Too many delete() calls in tests.
Solution:   Use deferred delete where possible.

brammool committed:
patch 9.0.0620: matchaddpos() can only add up to 8 matches

Problem:    matchaddpos() can only add up to 8 matches.
Solution:   Allocate the array of positions. (closes #11248)

dpelle committed:
patch 9.0.0621: filetype test leaves file behind

Problem:    Filetype test leaves file behind.
Solution:   Add deferred delete flag to writefile(). (Dominique Pellé,
            closes #11249)

brammool committed:
patch 9.0.0622: matchaddpos() can get slow when adding many matches

Problem:    matchaddpos() can get slow when adding many matches.
Solution:   Update the next available match ID when manually picking an ID and
            remove check if the available ID can be used. (idea by Rick Howe)

brammool committed:
patch 9.0.0623: error for modifying a const is not detected at compile time

Problem:    Error for modifying a const is not detected at compile time.
Solution:   Add TTFLAG_CONST and check for it in add() and extend().

brammool committed:
patch 9.0.0624: leaking argument type array

Problem:    Leaking argument type array.
Solution:   Add allocated memory to type_gap.

brammool committed:
patch 9.0.0625: too many delete() calls in tests

Problem:    Too many delete() calls in tests.
Solution:   Use deferred delete where possible.

brammool committed:
patch 9.0.0626: too many delete() calls in tests

Problem:    Too many delete() calls in tests.
Solution:   Use deferred delete where possible.

brammool committed:
patch 9.0.0627: "const" and "final" both make the type a constant

Problem:    "const" and "final" both make the type a constant. (Daniel
            Steinberg)
Solution:   Only have "const" make the type a constant.

brammool committed:
patch 9.0.0628: Coverity warns for not checking return value

Problem:    Coverity warns for not checking return value.
Solution:   Check the return value and simplify the code.

brammool committed:
patch 9.0.0629: get an error for using const only when executing

Problem:    Get an error for using const only when executing.
Solution:   Check for const at compile time for filter(), map(), remove(),
            reverse(), sort() and uniq().

brammool committed:
patch 9.0.0630: in Vim9 script a numbered function cannot be called

Problem:    In Vim9 script a numbered function cannot be called.
Solution:   Do not require "g:" before a numbered function name.
            (closes #11254)

brammool committed:
patch 9.0.0631: too many delete() calls in tests

Problem:    Too many delete() calls in tests.
Solution:   Use deferred delete where possible.

brammool committed:
patch 9.0.0632: calling a function from an "expr" option has overhead

Problem:    Calling a function from an "expr" option has too much overhead.
Solution:   Add call_simple_func() and use it for 'foldexpr'

rbtnn committed:
patch 9.0.0633: FEAT_TITLE was removed but is still used

Problem:    FEAT_TITLE was removed but is still used.
Solution:   Remove FEAT_TITLE. (Naruhiko Nishino, closes #11256)

brammool committed:
patch 9.0.0634: evaluating "expr" options has more overhead than needed

Problem:    Evaluating "expr" options has more overhead than needed.
Solution:   Use call_simple_func() for 'foldtext', 'includeexpr', 'printexpr',
            "expr" of 'spellsuggest', 'diffexpr', 'patchexpr', 'balloonexpr',
            'formatexpr', 'indentexpr' and 'charconvert'.

brammool committed:
patch 9.0.0635: build error and compiler warnings

Problem:    Build error and compiler warnings.
Solution:   Add missing change.  Add type casts.

brammool committed:
patch 9.0.0636: underline color may not work in some terminals

Problem:    Underline color does not work in terminals that don't send a
            termresponse.
Solution:   Do output t_8u if it was set explicitly. (closes #11253)

brammool committed:
patch 9.0.0637: syntax of commands in Vim9 script depends on +eval feature

Problem:    Syntax of commands in Vim9 script depends on +eval feature.
Solution:   Use same syntax with and without the +eval feature.

brammool committed:
patch 9.0.0638: popup menu highlight wrong on top of preview popup

Problem:    Popup menu highlight wrong on top of preview popup. (Yegappan
            Lakshmanan)
Solution:   Also check for the popup menu in screen_line().

brammool committed:
patch 9.0.0639: checking for popup in screen_char() is too late

Problem:    Checking for popup in screen_char() is too late, the attribute has
            already been changed.
Solution:   Move check for popup to where screen_char() is called.

brammool committed:
patch 9.0.0640: cannot scroll by screen line if a line wraps

Problem:    Cannot scroll by screen line if a line wraps.
Solution:   Add the 'smoothscroll' option.  Only works for CTRL-E and CTRL-Y
            so far.

brammool committed:
patch 9.0.0641: missing part of the new option code

Problem:    Missing part of the new option code.
Solution:   Add missing WV_SMS.

brammool committed:
patch 9.0.0642: breakindent test fails

Problem:    Breakindent test fails.
Solution:   Correct logic for resetting need_showbreak.

brammool committed:
patch 9.0.0643: smoothscroll test fails

Problem:    Smoothscroll test fails.
Solution:   Check if skipcol changed.

brammool committed:
patch 9.0.0644: 'smoothscroll' is not copied to a new window on :split

Problem:    'smoothscroll' is not copied to a new window on :split.
Solution:   Copy the option value.  Add a test.

brammool committed:
patch 9.0.0645: CTRL-Y does not stop at line 1

Problem:    CTRL-Y does not stop at line 1. (John Marriott)
Solution:   Stop at line 1 when 'smoothscroll' is not set. (closes #11261)

brammool committed:
patch 9.0.0646: with 'smoothscroll' CTRL-E is wrong when 'foldmethod' set

Problem:    with 'smoothscroll' set CTRL-E does not work properly when
            'foldmethod' is set to "indent". (Yee Cheng Chin)
Solution:   Merge the code for scroling with folds and 'smoothscroll'.
            (closes #11262)

luukvbaal committed:
patch 9.0.0647: the 'splitscroll' option is not a good name

Problem:    The 'splitscroll' option is not a good name.
Solution:   Rename 'splitscroll' to 'splitkeep' and make it a string option,
            also supporting "topline". (Luuk van Baal, closes #11258)

yegappan committed:
patch 9.0.0648: when using powershell input redirection does not work

Problem:    When using powershell input redirection does not work.
Solution:   Use a different shell command for powershell. (Yegappan
            Lakshmanan, closes #11257)

brammool committed:
patch 9.0.0649: no indication the first line is broken for 'smoothscroll'

Problem:    No indication when the first line is broken for 'smoothscroll'.
Solution:   Show "<<<" in the first line.

brammool committed:
patch 9.0.0650: some tests are failing

Problem:    Some tests are failing.
Solution:   Adjust for "<<<" showing up.

brammool committed:
patch 9.0.0651: build fails without the +conceal feature

Problem:    Build fails without the +conceal feature.
Solution:   Rename called function.

brammool committed:
Update runtime files

brammool committed:
patch 9.0.0652: 'smoothscroll' not tested with 'number' and "n" in 'cpo'

Problem:    'smoothscroll' not tested with 'number' and "n" in 'cpo'.
Solution:   Add tests, fix uncovered problem.

brammool committed:
patch 9.0.0653: BS and DEL do not work properly in an interacive shell

Problem:    BS and DEL do not work properly in an interacive shell. (Gary
            Johnson)
Solution:   Adjust the length for replaced codes.

brammool committed:
patch 9.0.0654: breakindent test fails

Problem:    Breakindent test fails.
Solution:   Temporarily accept wrong result.

brammool committed:
patch 9.0.0655: passing modifier codes to a shell running in the GUI

Problem:    passing modifier codes to a shell running in the GUI. (Gary
            Johnson)
Solution:   Include modifier codes into the key and drop the modifiers.

brammool committed:
patch 9.0.0656: cannot specify another character to use instead of '@'

Problem:    Cannot specify another character to use instead of '@' at the end
            of the window.
Solution:   Add "lastline" to 'fillchars'. (Martin Tournoij, closes #11264,
            closes #10963)

arp242 committed:
patch 9.0.0657: too many #ifdefs

Problem:    Too many #ifdefs.
Solution:   Graduate the +cmdwin feature.  Now the tiny and small builds are
            equal, drop the small build.  (Martin Tournoij, closes #11268)

brammool committed:
patch 9.0.0658: tiny build fails on Mac OS

Problem:    Tiny build fails on Mac OS.
Solution:   Define FEAT_CLIPBOARD only for normal build.

brammool committed:
patch 9.0.0659: wrong type of comment in SetSyn() function

Problem:    Wrong type of comment in SetSyn() function.
Solution:   Use Vim9 comment. (closes #11278)

zeertzjq committed:
patch 9.0.0660: mapping with CTRL keys does not work in the GUI

Problem:    Mapping with CTRL keys does not work in the GUI.
Solution:   Recognize CSI next to K_SPECIAL. (closes #11275, closes #11270)

zeertzjq committed:
patch 9.0.0661: multi-byte "lastline" in 'fillchars' does not work properly

Problem:    Multi-byte "lastline" item in 'fillchars' does not work properly
            when the window is two columns wide.
Solution:   Compute the text length correctly. (closes #11280)

brammool committed:
patch 9.0.0662: concealed characters do not work correctly

Problem:    Concealed characters do not work correctly.
Solution:   Subtract boguscols instead of adding them. (closes #11273)

zeertzjq committed:
patch 9.0.0663: tests check for +cmdwin feature which is always present

Problem:    Tests check for +cmdwin feature which is always present.
Solution:   Remove the checks. (closes #11287)

brammool committed:
patch 9.0.0664: bad redrawing with spell checking, using "C" and "$" in 'cpo'

Problem:    Bad redrawing with spell checking, using "C" and "$" in 'cpo'.
Solution:   Do not redraw the next line when "$" is in 'cpo'. (closes #11285)

brammool committed:
patch 9.0.0665: setting 'cmdheight' has no effect if last window was resized

Problem:    Setting 'cmdheight' has no effect if last window was resized.
Solution:   Do apply 'cmdheight' when told to.  Use the frame height instead
            of the cmdline_row.  (closes #11286)

brammool committed:
patch 9.0.0666: spacing-combining characters handled as composing

Problem:    Spacing-combining characters handled as composing, causing text to
            take more space than expected.
Solution:   Handle characters marked with "Mc" not as composing.
            (closes #11282

luukvbaal committed:
patch 9.0.0667: ml_get error when 'splitkeep' is "screen"

Problem:    ml_get error when 'splitkeep' is "screen". (Marius Gedminas)
Solution:   Check the botline is not too large. (Luuk van Baal,
            closes #11293, closes #11292)

pheiduck committed:
patch 9.0.0668: CI on Mac M1 only uses clang

Problem:    CI on Mac M1 only uses clang
Solution:   Also run with gcc. (closes #11263)

brammool committed:
patch 9.0.0669: too many delete() calls in tests

Problem:    Too many delete() calls in tests.
Solution:   Use deferred delete where possible.

brammool committed:
patch 9.0.0670: no space for command line when there is a tabline

Problem:    No space for command line when there is a tabline.
Solution:   Correct computation of where the command line should be.
            (closes #11295)

brammool committed:
patch 9.0.0671: negative topline using CTRL-Y with 'smoothscroll' and 'diff'

Problem:    Negative topline using CTRL-Y with 'smoothscroll' and 'diff'.
            (Ernie Rael)
Solution:   Only use 'smoothscroll' when 'wrap' is set.

brammool committed:
patch 9.0.0672: line partly shows with 'smoothscroll' and 'scrolloff' zero

Problem:    Cursor line only partly shows with 'smoothscroll' and 'scrolloff'
            zero.
Solution:   Do not use 'smoothscroll' when adjusting the bottom of the window.
            (closes #11269)

brammool committed:
patch 9.0.0673: first line wong with 'smoothscroll' and 'scrolloff' zero

Problem:    First line not scrolled properly with 'smoothscroll' and
            'scrolloff' zero and using "k".
Solution:   Make sure the cursor position is visible.

brammool committed:
patch 9.0.0674: build error with tiny version

Problem:    Build error with tiny version.
Solution:   Use PLINES_NOFILL macro.

brammool committed:
patch 9.0.0675: search test screendump is outdated

Problem:    Search test screendump is outdated.
Solution:   Update the screendump for improved display.

ichizok committed:
patch 9.0.0676: CI on Mac M1 with gcc actually uses clang

Problem:    CI on Mac M1 with gcc actually uses clang.
Solution:   Remove the gcc task. (Ozaki Kiichi, closes #11297)

brammool committed:
patch 9.0.0677: breakindent test accepts wrong result

Problem:    Breakindent test accepts wrong result.
Solution:   Fix the number column and adjust the expected text.

brammool committed:
patch 9.0.0678: using exclamation marks on :function

Problem:    Using exclamation marks on :function.
Solution:   Use :func and :endfunc as usual.

brammool committed:
patch 9.0.0679: tests failing with 'smoothscroll', 'number' and "n" in 'cpo'

Problem:    Tests failing with 'smoothscroll', 'number' and "n" in 'cpo'.
Solution:   Do not count number column in topline if columns are skipped.

brammool committed:
patch 9.0.0680: tests failing with 'breakindent', 'number' and "n" in 'cpo'

Problem:    Tests failing with 'breakindent', 'number' and "n" in 'cpo'.
Solution:   Do count the number column in topline if 'breakindent' is set.

brammool committed:
patch 9.0.0681: "<<<" shows for 'smoothscroll' even when 'showbreak is set

Problem:    "<<<" shows for 'smoothscroll' even when 'showbreak is set.
Solution:   When 'showbreak' is set do not display "<<<".

brammool committed:
patch 9.0.0682: crash when popup with deleted timer is closed

Problem:    Crash when popup with deleted timer is closed. (Igbanam
            Ogbuluijah)
Solution:   Check the timer still exists. (closes #11301)

brammool committed:
patch 9.0.0683: cannot specify a time for :echowindow

Problem:    Cannot specify a time for :echowindow.
Solution:   A count can be used to specify the display time. Add
            popup_findecho().

brammool committed:
patch 9.0.0684: skipped :exe command fails compilation on MS-Windows

Problem:    Skipped :exe command fails compilation on MS-Windows.
Solution:   Adjust return value when skipping.

ychin committed:
patch 9.0.0685: FORTIFY_SOURCE causes a crash in Vim9 script

Problem:    FORTIFY_SOURCE causes a crash in Vim9 script.
Solution:   Use a pointer to the first char. (Yee Cheng Chin, closes #11302)

haron13-2019 committed:
patch 9.0.0686: the right ALT key does not work on some MS-Windows keyboards

Problem:    The right ALT key does not work on some MS-Windows keyboards.
Solution:   Adjust the modifiers based on GetKeyState(). (Anoton Sharonov,
            closes #11300)

brammool committed:
patch 9.0.0687: "export def" does not work in a nested block

Problem:    "export def" does not work in a nested block.
Solution:   Do not handle "export" with a separate function but in the same
            command stack. (closes #11304)

brammool committed:
patch 9.0.0688: debugger does not display the whole command

Problem:    Debugger does not display the whole command.
Solution:   Set ea.cmd before checking for a breakpoint.

brammool committed:
patch 9.0.0689: compiler warning for unused function

Problem:    Compiler warning for unused function.
Solution:   Add #ifdef. (John Marriott)

brammool committed:
patch 9.0.0690: buffer size for expanding tab not correctly computed

Problem:    Buffer size for expanding tab not correctly computed.
Solution:   Correctly use size of end character.

brammool committed:
patch 9.0.0691: lalloc(0) error in listchars test

Problem:    lalloc(0) error in listchars test.
Solution:   Skip generating text for tab if tab_len is zero.

ObserverOfTime committed:
patch 9.0.0692: PoE filter files are not recognized

Problem:    PoE filter files are not recognized.
Solution:   Add a pattern to detect PoE filter files. (closes #11305)

brammool committed:
patch 9.0.0693: browse() first argument cannot be a bool

Problem:    browse() first argument cannot be a bool.
Solution:   Use tv_get_bool_chk() instead of tv_get_number_chk().
            (closes #11308)

ychin committed:
patch 9.0.0694: no native sound support on Mac OS

Problem:    No native sound support on Mac OS.
Solution:   Add sound support for Mac OS. (Yee Cheng Chin, closes #11274)

brammool committed:
patch 9.0.0695: failing check for dictionary type for const any

Problem:    Failing check for dictionary type for const any.
Solution:   Check for any type properly. (closes #11310)

brammool committed:
patch 9.0.0696: it is unclear if the +rightleft and +arabic features are used

Problem:    It is unclear if the +rightleft and +arabic features are actively
            being used.
Solution:   Disable the features, await feedback.

brammool committed:
patch 9.0.0697: cursor in wrong position with Visual substitute

Problem:    Cursor in wrong position with Visual substitute.
Solution:   When restoring 'linebreak' mark the virtual column as invalid.
            (closes #11309, closes #11311)

arp242 committed:
patch 9.0.0698: VisVim is outdated, does not work with current Visual Studio

Problem:    VisVim is outdated, does not work with current Visual Studio.
Solution:   Remove VisVim. (Martin Tournoij)

brammool committed:
patch 9.0.0699: tiny build fails

Problem:    Tiny build fails.
Solution:   Add #ifdef.

arp242 committed:
patch 9.0.0700: there is no real need for a "big" build

Problem:    There is no real need for a "big" build.
Solution:   Move common features to "normal" build, less often used features
            to the "huge" build. (Martin Tournoij, closes #11283)

brammool committed:
patch 9.0.0701: with 'smoothscroll' cursor position not adjusted in long line

Problem:    With 'smoothscroll' the cursor position s not adjusted in a long
            line.
Solution:   Move the cursor further up or down in the line.

zeertzjq committed:
patch 9.0.0702: incomplete testing cursor position with 'linebreak' set

Problem:    Incomplete testing cursor position after change with 'linebreak'
            set.
Solution:   Add a test and move test cases together. (closes #11313)

brammool committed:
patch 9.0.0703: failing check for argument type for const any

Problem:    Failing check for argument type for const any.
Solution:   Check for any type properly. (closes #11316)

rbtnn committed:
patch 9.0.0704: CI runs "tiny" and "small" builds, which are the same

Problem:    CI runs "tiny" and "small" builds, which are the same.
Solution:   Remove the "small" build. (Naruhiko Nishino, closes #11315)

brammool committed:
patch 9.0.0705: virtual text truncation does not take padding into account

Problem:    Virtual text truncation does not take padding into account.
Solution:   Subtract the padding from the available space. (closes #11318)

brammool committed:
patch 9.0.0706: :help in a narrow window always opens at the top

Problem:    :help in a narrow window always opens at the top.
Solution:   Respect 'splitbelow'. (closes #11319)

brammool committed:
patch 9.0.0707: with 'smoothscroll' cursor position not adjusted in long line

Problem:    With 'smoothscroll' and 'scrolloff' non-zero the cursor position
            is not properly adjusted in a long line.
Solution:   Move the cursor further up or down in the line.

ychin committed:
patch 9.0.0708: :confirm does not work properly for a terminal buffer

Problem:    :confirm does not work properly for a terminal buffer.
Solution:   Handle :confirm for a terminal buffer differently.  (Yee Cheng
            Chin, closes #11312)

brammool committed:
patch 9.0.0709: virtual text "after" not correct with 'nowrap'

Problem:    Virtual text "after" not correct with 'nowrap'.
Solution:   Do not display "after" text prop on the next line when 'wrap' is
            off.

ychin committed:
patch 9.0.0710: quitting/unloading/hiding a terminal does not work properly

Problem:    Quitting/unloading/hiding a terminal buffer does not always work
            properly.
Solution:   Avoid that ":q!" leaves an empty buffer behind.  ":bunload!" also
            kills the job and unloads the buffer.  ":hide" does not unload the
            buffer. (Yee Cheng Chin, closes #11323)

ObserverOfTime committed:
patch 9.0.0711: SubStation Alpha files are not recognized

Problem:    SubStation Alpha files are not recognized.
Solution:   Add patterns for SubStation Alpha files. (closes #11332)

brammool committed:
patch 9.0.0712: wrong column when calling setcursorcharpos() with zero lnum

Problem:    Wrong column when calling setcursorcharpos() with zero lnum.
Solution:   Set the line number before calling buf_charidx_to_byteidx().
            (closes #11329)

zeertzjq committed:
patch 9.0.0713: <amatch> of MenuPopup event is expanded like a file name

Problem:    <amatch> of MenuPopup event is expanded like a file name.
Solution:   Do not expand <amatch> for MenuPopup. (closes #11328)

brammool committed:
patch 9.0.0714: with 'nowrap' two virtual text below not displayed correctly

Problem:    With 'nowrap' two virtual text below not displayed correctly.
Solution:   Set text_prop_follows before continuing.  Correct for number
            column. (closes #11333)

brammool committed:
patch 9.0.0715: wrong argument for append() gives two error messages

Problem:    Wrong argument for append() gives two error messages.
Solution:   When getting an error for a number argument don't try using it as
            a string. (closes #11335)

brammool committed:
patch 9.0.0716: with 'nowrap' virtual text "after" does not scroll left

Problem:    With 'nowrap' virtual text "after" does not scroll left.
Solution:   Skip part of the virtual text that is left of the window.
            (closes #11320)  Fix going beyond the last column of the window.

brammool committed:
patch 9.0.0717: compiler warning for unused variable in tiny build

Problem:    Compiler warning for unused variable in tiny build.
Solution:   Add #ifdefs.

brammool committed:
patch 9.0.0718: extra empty line between two virtual text "below"

Problem:    Extra empty line between two virtual text "below" when 'wrap' and
            'number' are set.
Solution:   Reset "before" when there is no text in the screen line.
            (closes #11334)

brammool committed:
patch 9.0.0719: too many delete() calls in tests

Problem:    Too many delete() calls in tests.
Solution:   Use deferred delete where possible.

brammool committed:
patch 9.0.0720: MS-Windows GUI may have pixel dust from antialiasing

Problem:    MS-Windows GUI may have pixel dust from antialiasing.
Solution:   When a character changes also redraw the next one. (issue #8532)

brammool committed:
patch 9.0.0721: virtual text "above" with padding not displayed correctly

Problem:    Virtual text "above" with padding not displayed correctly.
Solution:   Take padding into account when truncating. (closes #11340)

brammool committed:
patch 9.0.0722: virtual text "after" does not show with 'list' set

Problem:    Virtual text "after" does not show with 'list' set.
Solution:   Do not break out of the loop when another text prop follows.
            (closes #11337)

brammool committed:
patch 9.0.0723: extra empty line below virtual text when 'list' is set

Problem:    Extra empty line below virtual text when 'list' is set.
Solution:   Do not reset lcs_eol_one but set text_prop_follows. (closes #11339)

brammool committed:
patch 9.0.0724: closure in compiled function gets same variable in block

Problem:    Closure in compiled function gets same variable in block.
Solution:   At the end of a block to not always reset the variable count.
            (issue #11094)

brammool committed:
patch 9.0.0725: virtual text "after" wraps to next line when 'wrap' is off

Problem:    Virtual text "after" wraps to next line even when 'wrap' is off
            and 'list' is set.
Solution:   Do not use the minimum width when 'wrap' is off. (issue #11336)

brammool committed:
patch 9.0.0726: looping over list of lists works in script, not in function

Problem:    Looping over list of lists and changing the list contents works in
            Vim9 script, not in a compiled function.
Solution:   Mark the loop variable final instead of const. (closes #11347)

brammool committed:
patch 9.0.0727: help in the repository differs from patched version too much

Problem:    Help in the repository differs from patched version too much.
Solution:   Make a patch for a few help files.

brammool committed:
patch 9.0.0728: extend() test fails

Problem:    extend() test fails.
Solution:   Item is final, not const.

brammool committed:
patch 9.0.0729: the rightleft and arabic features are disabled

Problem:    The rightleft and arabic features are disabled.
Solution:   Re-enable the features, some users want to use the functionality.

brammool committed:
patch 9.0.0730: startup test fails with right-left feature

Problem:    Startup test fails with right-left feature.
Solution:   Do not delete test file too early.

dundargoc committed:
patch 9.0.0731: clang-tidy configuration files are not recognized

Problem:    clang-tidy configuration files are not recognized.
Solution:   Recognize clang-tidy files as yaml. (closes #11350)

brammool committed:
patch 9.0.0732: no check for white space before and after "=<<"

Problem:    No check for white space before and after "=<<". (Doug Kearns)
Solution:   Check for white space in Vim9 script. (closes #11351)

dpelle committed:
patch 9.0.0733: use of strftime() is not safe

Problem:    Use of strftime() is not safe.
Solution:   Check the return value of strftime().  Use a larger buffer and
            correctly pass the available space. (Dominique Pellé, closes
            #11348)

brammool committed:
patch 9.0.0734: cursor position invalid when scrolling with 'smoothscroll'

Problem:    Cursor position invalid when scrolling with 'smoothscroll' set.
            (Ernie Rael)
Solution:   Add w_valid_skipcol and clear flags when it changes.  Adjust
            w_skipcol after moving the cursor.

brammool committed:
patch 9.0.0735: breakindent and scrolloff tests fail

Problem:    Breakindent and scrolloff tests fail.
Solution:   Temporarily skip the assertions.

yegappan committed:
patch 9.0.0736: quickfix listing does not handle very long messages

Problem:    Quickfix listing does not handle very long messages.
Solution:   Use a growarray instead of a fixed size buffer. (Yegappan
            Lakshmanan, closes #11357)

brammool committed:
patch 9.0.0737: Lisp word only recognized when a space follows

Problem:    Lisp word only recognized when a space follows.
Solution:   Also match a word at the end of a line.  Rename the test.  Use a
            compiled function to avoid backslashes.

bfredl committed:
patch 9.0.0738: cannot suppress completion "scanning" messages

Problem:    Cannot suppress completion "scanning" messages.
Solution:   Add the "C" flag in 'shortmess'. (Bjorn Linse, closes #11354)

ychin committed:
patch 9.0.0739: mouse column not correctly used for popup_setpos

Problem:    Mouse column not correctly used for popup_setpos.
Solution:   Adjust off-by-one error and handle Visual line selection properly.
            (Yee Cheng Chin, closes #11356)

brammool committed:
patch 9.0.0740: prop_add_list() gives multiple errors for invalid argument

Problem:    prop_add_list() gives multiple errors for invalid argument.
Solution:   Only give one error message.

brammool committed:
patch 9.0.0741: cannot specify an ID for each item with prop_add_list()

Problem:    Cannot specify an ID for each item with prop_add_list(). (Sergey
            Vlasov)
Solution:   Add an optional fifth number to the item. (closes #11360)

brammool committed:
patch 9.0.0742: reading past end of the line when compiling a function

Problem:    Reading past end of the line when compiling a function with
            errors.
Solution:   Do not return an invalid pointer.  Fix skipping redirection.

brammool committed:
patch 9.0.0743: starting cscope on Unix does not quote the arguments right

Problem:    Starting cscope on Unix does not quote the arguments correctly.
            (Gary Johnson)
Solution:   Move the final quote after the arguments.

brammool committed:
patch 9.0.0744: in script in autoload dir exported variable is not found

Problem:    In script in autoload dir exported variable is not found. (Doug
            Kearns)
Solution:   Find the variable with the "script#" prefix. (closes #11361)

brammool committed:
patch 9.0.0745: wrong cursor position when using "gj" and "gk" in a long line

Problem:    Wrong cursor position when using "gj" and "gk" in a long line.
Solution:   Adjust computations for the cursor position and skipcol. Re-enable
            tests that pass now, disable failing breakindent test.

brammool committed:
patch 9.0.0746: breakindent test cases are commented out

Problem:    Breakindent test cases are commented out.
Solution:   Adjust expected result to slightly different behavior.  Correct
            computations for cursor position.

arp242 committed:
patch 9.0.0747: too many #ifdefs

Problem:    Too many #ifdefs.
Solution:   Gradudate the +cmdline_info feature. (Martin Tournoij,
            closes #11330)

trygveaa committed:
patch 9.0.0748: Kitty may send key without modifiers with CSI u code

Problem:    Kitty may send key without modifiers with CSI u code.
Solution:   Handle CSI u code without modifiers. (Trygve Aaberge,
            closes #11364)

yegappan committed:
patch 9.0.0749: alloc/free of buffer for each quickfix entry is inefficient

Problem:    Alloc/free of buffer for each quickfix entry is inefficient.
Solution:   Use a shared grow array. (Yegappan Lakshmanan, closes #11365)

brammool committed:
patch 9.0.0750: crash when popup closed in callback

Problem:    Crash when popup closed in callback. (Maxim Kim)
Solution:   In syntax_end_parsing() check that syn_block is valid.

brammool committed:
patch 9.0.0751: 'scrolloff' does not work well with 'smoothscroll'

Problem:    'scrolloff' does not work well with 'smoothscroll'.
Solution:   Make positioning the cursor a bit better.  Rename functions.

ObserverOfTime committed:
patch 9.0.0752: Rprofile files are not recognized

Problem:    Rprofile files are not recognized.
Solution:   Recognize Rprofile files as "r". (closes #11369)

ychin committed:
patch 9.0.0753: some Ex commands are not in the help index

Problem:    Some Ex commands are not in the help index.
Solution:   Add the missing commands.  Add a script to check all Ex commands
            are in the help index. (Yee Cheng Chin, closes #11371)

brammool committed:
patch 9.0.0754: 'indentexpr' overrules lisp indenting in one situation

Problem:    'indentexpr' overrules lisp indenting in one situation.
Solution:   Add "else" to keep the lisp indent. (issue #11327)

brammool committed:
patch 9.0.0755: huge build on macos always fails on CI

Problem:    Huge build on macos always fails on CI.
Solution:   Temporarily disable the perl interface.

Shougo committed:
patch 9.0.0756: no autocmd event for changing text in a terminal window

Problem:    No autocmd event for changing text in a terminal window.
Solution:   Add TextChangedT. (Shougo Matsushita, closes #11366)

brammool committed:
patch 9.0.0757: line number not visisble with 'smoothscroll', 'nu' and 'rnu'

Problem:    Line number not visisble with 'smoothscroll', 'nu' and 'rnu'.
Solution:   Put the ">>>" after the line number instead of on top.

brammool committed:
patch 9.0.0758: "precedes" from 'listchars' overwritten by <<<

Problem:    "precedes" from 'listchars' overwritten by <<< for 'smoothscroll'.
Solution:   Keep the "precedes" character.

pheiduck committed:
patch 9.0.0759: huge build on macos does not use Perl

Problem:    Huge build on macos does not use Perl.
Solution:   Re-enable the Perl interface using "dynamic". (closes #11375)

brammool committed:
patch 9.0.0760: display test for 'listchars' "precedes" fails

Problem:    Display test for 'listchars' "precedes" fails.
Solution:   Correct the expected result.

brammool committed:
patch 9.0.0761: cannot use 'indentexpr' for Lisp indenting

Problem:    Cannot use 'indentexpr' for Lisp indenting.
Solution:   Add the 'lispoptions' option.

brammool committed:
patch 9.0.0762: build failure

Problem:    Build failure.
Solution:   Add missing change.

brammool committed:
patch 9.0.0763: MS-Windows: warning for using int for size_t

Problem:    MS-Windows: warning for using int for size_t.
Solution:   Declare variable as size_t.

brammool committed:
patch 9.0.0764: indent and option tests fail

Problem:    Indent and option tests fail.
Solution:   Change OP_INDENT.  Add entry to options test table.

brammool committed:
patch 9.0.0765: with a Visual block a put command column may go negative

Problem:    With a Visual block a put command column may go negative.
Solution:   Check that the column does not become negative.

brammool committed:
patch 9.0.0766: too many delete() calls in tests

Problem:    Too many delete() calls in tests.
Solution:   Use deferred delete where possible.

brammool committed:
patch 9.0.0767: too many delete() calls in tests

Problem:    Too many delete() calls in tests.
Solution:   Use deferred delete where possible.

brammool committed:
patch 9.0.0768: too many delete() calls in tests

Problem:    Too many delete() calls in tests.
Solution:   Use deferred delete where possible.

brammool committed:
patch 9.0.0769: too many delete() calls in tests

Problem:    Too many delete() calls in tests.
Solution:   Use deferred delete where possible.

yegappan committed:
patch 9.0.0770: quickfix commands may keep memory allocated

Problem:    Quickfix commands may keep memory allocated.
Solution:   Free memory when it's a bit much. (Yegappan Lakshmanan,
            closes #11379)

arp242 committed:
patch 9.0.0771: cannot always tell the difference beween tex and rexx files

Problem:    Cannot always tell the difference beween tex and rexx files.
Solution:   Recognize tex by a leading backslash. (Martin Tournoij,
            closes #11380)

brammool committed:
Update runtime files

brammool committed:
patch 9.0.0772: the libvterm code is outdated

Problem:    The libvterm code is outdated.
Solution:   Include libvterm changes from revision 790 to 801.

pheiduck committed:
patch 9.0.0773: huge build on macos uses dynamic Perl

Problem:    Huge build on macos uses dynamic Perl.
Solution:   Use built-in Perl, uninstall the brew one. (closes #11382)

brammool committed:
patch 9.0.0774: the libvterm code is outdated

Problem:    The libvterm code is outdated.
Solution:   Include libvterm changes from revision 802 to 817.  Revert some
            changes made for C89.

zewpo committed:
patch 9.0.0775: MS-Windows: mouse scrolling not supported in the console

Problem:    MS-Windows: mouse scrolling not supported in the console.
Solution:   Add event handling for mouse scroll events. (Christopher
            Plewright, closes #11374)

brammool committed:
patch 9.0.0776: MSVC can't have field name "small"

Problem:    MSVC can't have field name "small".
Solution:   Rename small to smallfont.

yegappan committed:
patch 9.0.0777: code is indented too much

Problem:    Code is indented too much.
Solution:   Use an early return. (Yegappan Lakshmanan, closes #11386)

brammool committed:
patch 9.0.0778: indexing of unknown const type fails during compilation

Problem:    Indexing of unknown const type fails during compilation.
Solution:   Check for "any" properly. (closes #11389)

dkearns committed:
patch 9.0.0779: lsl and lm3 file extensions are not recognized

Problem:    lsl and lm3 file extensions are not recognized.
Solution:   Add *.lsl and *.lm3 patterns. (Doug Kearns, closes #11384)

luukvbaal committed:
patch 9.0.0780: 'scroll' value computed in unexpected location

Problem:    'scroll' value computed in unexpected location.
Solution:   Compute 'scroll' when the window height is changed. (Luuk van
            Baal, closes #11387)

brammool committed:
patch 9.0.0781: workaround to rename "small" to "smallfont" is clumsy

Problem:    Workaround to rename "small" to "smallfont" is clumsy.
Solution:   Undefine "small" after including windows.h. (Ken Takata)

ObserverOfTime committed:
patch 9.0.0782: OpenVPN files are not recognized

Problem:    OpenVPN files are not recognized.
Solution:   Add patterns for OpenVPN files. (closes #11391)

brammool committed:
patch 9.0.0783: ":!" doesn't do anything but does update the previous command

Problem:    ":!" doesn't do anything but does update the previous command.
Solution:   Do not have ":!" change the previous command. (Martin Tournoij,
            closes #11372)

brammool committed:
patch 9.0.0784: text prop "above" not right with 'number' and "n" in 'cpo'

Problem:    Text prop "above" not displayed correctly with 'number' and "n" in
            'cpo'.
Solution:   Draw the line number column until the line text is reached.

brammool committed:
patch 9.0.0785: memory leak with empty shell command

Problem:    Memory leak with empty shell command.
Solution:   Free the allocated memory when bailing out.

zeertzjq committed:
patch 9.0.0786: user command does not get number from :tab modifier

Problem:    User command does not get number from :tab modifier.
Solution:   Include the number. (closes #11393, closes #6801)

zewpo committed:
patch 9.0.0787: mouse scrolling in terminal misbehaves without dll

Problem:    MS-Windows: mouse scrolling in terminal misbehaves without dll.
Solution:   Add #ifdef as a temporary solution. (Christopher Plewright,
            closes #11392)

brammool committed:
patch 9.0.0788: ModeChanged autocmd not executed when Visual ends with CTRL-C

Problem:    ModeChanged autocmd not executed when Visual mode is ended with
            CTRL-C.
Solution:   Do not trigger the autocmd when got_int is set. (closes #11394)

brammool committed:
patch 9.0.0789: dummy buffer ends up in a window

Problem:    Dummy buffer ends up in a window.
Solution:   Disallow navigating to a dummy buffer.

brammool committed:
patch 9.0.0790: test for dummy buffer does not always produce the E86 error

Problem:    Test for dummy buffer does not always produce the E86 error.
Solution:   Do not check if the error is produced.

trygveaa committed:
patch 9.0.0791: at the hit-Enter prompt the End and Home keys may not work

Problem:    At the hit-Enter prompt the End and Home keys may not work.
Solution:   Use the special "@" code for End and Home, like it was done for
            the cursor keys in patch 8.2.2246. (Trygve Aaberge, closes #11396)

brammool committed:
patch 9.0.0792: MS-Windows: compiler complains about unused function

Problem:    MS-Windows: compiler complains about unused function.
Solution:   Add #ifdef. (John Marriott)

zewpo committed:
patch 9.0.0793: MS-Windows: mouse scroll events only work with the dll

Problem:    MS-Windows: mouse scroll events only work with the dll.
Solution:   Accept CSI codes for MS-Windows without the GUI. (Christopher
            Plewright, closes #11401)

brammool committed:
patch 9.0.0794: there is no way to find out if modifyOtherKeys has been seen

Problem:    There is no way to find out if an escape sequence with
            modifyOtherKeys has been seen.
Solution:   Add a notice with ":verbose map".

k-takata committed:
patch 9.0.0795: readblob() always reads the whole file

Problem:    readblob() always reads the whole file.
Solution:   Add arguments to read part of the file. (Ken Takata,
            closes #11402)

brammool committed:
patch 9.0.0796: mapping test fails in some situations

Problem:    Mapping test fails in some situations.
Solution:   Find the line with the verbose information.

zeertzjq committed:
patch 9.0.0797: order of assert function arguments is reverted

Problem:    Order of assert function arguments is reverted.
Solution:   Swap the arguments. (closes #11399)

marwing committed:
patch 9.0.0798: clang format configuration files are not recognized

Problem:    Clang format configuration files are not recognized.
Solution:   Use yaml for Clang format configuration files. (Marwin Glaser,
            closes #11398)

brammool committed:
patch 9.0.0799: in compiled function ->() on next line not recognized

Problem:    In compiled function ->() on next line not recognized.
Solution:   Also check for "(". (closes #11405)

brammool committed:
patch 9.0.0800: compiler complains about repeated typedef

Problem:    Compiler complains about repeated typedef.
Solution:   Remove one typedef.

brammool committed:
patch 9.0.0801: the modifyOtherKeys flag is set when it should not

Problem:    The modifyOtherKeys flag is set when it should not.
Solution:   Do not handle special key codes with a modifer value above 16 as a
            modifyOtherKeys value. (issue #11403)

zewpo committed:
patch 9.0.0802: MS-Windows: cannot map console mouse scroll events

Problem:    MS-Windows: cannot map console mouse scroll events.
Solution:   Change CSI to K_SPECIAL when checking for a mapping. (Christopher
            Plewright, closes #11410)

k-takata committed:
patch 9.0.0803: readblob() cannot read from character device

Problem:    readblob() cannot read from character device.
Solution:   Use S_ISCHR() to not check the size. (Ken Takata, closes #11407)

brammool committed:
patch 9.0.0804: crash when trying to divide a number by -1

Problem:    Crash when trying to divice the largest negative number by -1.
Solution:   Handle this case specifically.

brammool committed:
patch 9.0.0805: filetype autocmd may cause freed memory access

Problem:    Filetype autocmd may cause freed memory access.
Solution:   Set the quickfix-busy flag while filling the buffer.

zeertzjq committed:
patch 9.0.0806: 'langmap' works differently when there are modifiers

Problem:    'langmap' works differently when there are modifiers.
Solution:   Only apply 'langmap' to a character where modifiers have no
            effect. (closes #11395, closes #11404)

brammool committed:
patch 9.0.0807: with 'smoothscroll' typing "0" may not go to the first column

Problem:    With 'smoothscroll' typing "0" may not go to the first column.
Solution:   Recompute w_cline_height when needed.  Do not scroll up when it
            would move the cursor.

maxbrunet committed:
patch 9.0.0808: jsonnet filetype detection has a typo

Problem:    jsonnet filetype detection has a typo.
Solution:   Change "libjsonnet" to "libsonnet". (Maxime Brunet, closes #11412)

brammool committed:
patch 9.0.0809: test for job writing to buffer fails

Problem:    Test for job writing to buffer fails.
Solution:   Correct w_topline when deleting a buffer line.

brammool committed:
patch 9.0.0810: readblob() returns empty when trying to read too much

Problem:    readblob() returns empty when trying to read too much.
Solution:   Return what is available.

brammool committed:
patch 9.0.0811: error if :echowin is preceded by a command modifier

Problem:    Error if :echowin is preceded by a command modifier.
Solution:   Do not give an error for range when there is a modifier.
            (closes #11414)

zewpo committed:
patch 9.0.0812: GUI mouse scrollwheel mappings don't work

Problem:    GUI mouse scrollwheel mappings don't work.
Solution:   Add check for "gui.in_use". (Christopher Plewright, closes #11418)

brammool committed:
patch 9.0.0813: Kitty terminal is not recognized

Problem:    Kitty terminal is not recognized.
Solution:   Recognize Kitty by the termresponse and then do not set
            seenModifyOtherKeys, since Kitty doesn't support that.
            (issue #11413)

justinmk committed:
patch 9.0.0814: aws config files are not recognized

Problem:    Aws config files are not recognized.
Solution:   Use "confini" for aws config files. (Justin M. Keyes,
            closes #11416)

arp242 committed:
patch 9.0.0815

brammool committed:
patch 9.0.0817

brammool committed:
patch 9.0.0818

brammool committed:
Revert unintended Makefile change

brammool committed:
Fix wrong struct access for member.

brammool committed:
Add missing entry for the 'endoffile' option.

brammool committed:
Update runtime files

brammool committed:
patch 9.0.0819

brammool committed:
patch 9.0.0820: memory leak with empty shell command

Problem:    Memory leak with empty shell command.
Solution:   Free the empty string.

litoj committed:
i3config syntax with better type distinction

This syntax uses a couple more colour highlights to match the actual function of used keywords in the config file. This way it is easier to figure out what does what.
- distinctly highlights key combos to clearly differentiate it from the rest of the command
- all keywords that can be used as command line commands are highlighted as Functions
- conditions are now properly highlighted with all operators clearly visible
+ some bug fixed

litoj committed:
made adjustments to suit ongoing pull request

litoj committed:
Update i3config.vim

litoj committed:
Merge branch 'patch-1' of github.com:JosefLitos/vim into patch-1

litoj committed:
i3config: fixed string within string highlight
